### PR TITLE
add tests for libtensorflow and libtensorflow_cc

### PR DESCRIPTION
**Pull request by njzjz**
based on #87 and following @hajapy's advice 

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by njzjz** [comment]
@conda-forge-admin, please rerender

**Comment by njzjz** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -188,12 +193,18 @@ outputs:
...
         - libprotobuf
         - nsync
     test:
+      files:
+        - test_cc.cc
+      requires:
+        - {{ compiler('cxx') }}
       commands:
         - test -f $PREFIX/lib/libtensorflow_cc.so  # [not win]
+        - $CXX -std=c++11 -o test_cc -L${PREFIX}/lib/ -ltensorflow_cc -ltensorflow_framework -I${PREFIX}/include/ test_cc.cc  # [not win]
```
Comment (njzjz) [inline recipe/meta.yaml]:
```suggestion
        - $CXX -std=c++11 -o test_cc -L${PREFIX}/lib/ -ltensorflow_cc -ltensorflow_framework -lrt -I${PREFIX}/include/ test_cc.cc  # [not win]
```

**Comment by njzjz** [comment]
I am happy to see all checks passed after I tried many times to add tests for libtensorflow and libtensorflow_cc. I will merge this PR then.

**Comment by hajapy** [comment]
I’m glad this passed but I’m not sure about the libprotobuf change, won’t that make this package uninstallable with many other conda forge packages?

**Comment by hajapy** [comment]
For future reference, please ping me or others for a review before merging the PRs when they involve non-trivial changes.

**Comment by njzjz** [comment]
> I’m glad this passed but I’m not sure about the libprotobuf change, won’t that make this package uninstallable with many other conda forge packages?

TensorFlow downloads protobuf 3.6 as defined in [workspace.bzl](https://github.com/tensorflow/tensorflow/blob/v1.13.1/tensorflow/workspace.bzl), and builds with what it downloads. I tried to add `protobuf_archive` to TF_SYSTEM_LIBS but there seems to be a bug. So the version of libprotobuf by which TensorFlow built was 3.6 but not 3.8.

**Comment by njzjz** [comment]
I am not sure what the effect of `com_google_protobuf,com_google_protobuf_cc` is in `TF_SYSTEM_LIBS`, but in fact, it still needs libprotobuf 3.6 headers.

**Comment by njzjz** [comment]
> For future reference, please ping me or others for a review before merging the PRs when they involve non-trivial changes.

@hajapy I’ll pay attention to it.

**Comment by hajapy** [comment]
I think, though not positive, the libprotobuf 3.6 pinning should only be applied to the libtensorflow_cc output. The python tensorflow package should be allowed to work with any protobuf >=3.6.1 and not care about libprotobuf.

**Comment by jjhelmus** [comment]
> I will try to remove it.

Removing the run requirement will not be sufficient.  libprotobuf has a run_export which will add the appropriate requirements to the run requirements automatically as dictated by the ABI.  

Allowing bazel to build protobuf independently results in a static version of the library being included and the only run requirement is on the python binding, protobuf>=3.6.1.  This method if preferred.

**Pull request merged**